ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactionQueryBuilder.java

```diff
@@ -180,6 +188,9 @@ CompactionQueryBuilder setPartitioned(boolean partitioned) {
    * If true, Create operations for CRUD minor compaction will result in a bucketed table.
    */
   CompactionQueryBuilder setBucketed(boolean bucketed) {
+    if(rebalance && bucketed) {
+      throw new IllegalArgumentException("Rebalance compaction is supported only on non-bucketed tables!");
```

Comment:
The variable name is fine I think, it means if the table is explicitly bucketed or not. I added the check because rebalance compaction is not allowed for explicitly bucketed table. I changed the message to contain both words. :)